### PR TITLE
Feature/107 Unsubscribe from emails

### PR DESCRIPTION
**Pull request by evgenyneu**
Added ability for users to unsubscribe from emails (#107).

* Added Email Subscription section to settings:

settings

* After clicking "Unsubscribe" button, show the `/account/settings/change_email_subscription` page

Screenshot 2024-08-01 at 10 46 05 PM

* Clicking the button shows the success message

Screenshot 2024-08-01 at 10 46 10 PM

* For unsubscribed users the settings section looks like this

Screenshot 2024-08-01 at 10 47 39 PM

* Added Unsubscribe link to the welcome email, which points to `{WebsiteBaseUrl}/account/settings/change_email_subscription`:

Screenshot 2024-08-01 at 10 52 55 PM

text:

> Welcome to SaaS Starter! 
> This is a quick sample of a welcome email. You can customize this email to fit your needs.
> To unsubscribe, visit: https://saasstarter.work/account/settings/change_email_subscription

**Issue to be fixed**: If user it not logged in they will not be redirected to `/account/settings/change_email_subscription` after login, this could be implemented later?

## Summary by CodeRabbit

- **New Features**
	- Added functionality for users to manage their email subscription preferences directly from their account settings.
	- Introduced an unsubscribe link in welcome emails to allow users easy access to update their subscription status.

- **Bug Fixes**
	- Enhanced error handling when toggling email subscription status to ensure smoother user experience.

- **Documentation**
	- Updated README to provide clearer setup instructions for new and existing Supabase projects.

- **Tests**
	- Added comprehensive unit tests for the email subscription management functionality to ensure reliability and correctness.

### CODE REVIEW

**Comment by scosman** [comment]
Thanks! I’m a bit slammed but will take a look when I have some time!

**Comment by scosman** [comment]
Really fantastic PR. Thanks @evgenyneu.

Merging as is. Few nitpick things like wording, but those are optional and can come after. I might quickly do them now.

File: src/routes/(admin)/account/(menu)/settings/change_email_subscription/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+</h1>
+
+<SettingsModule
+  editable={true}
+  saveButtonTitle={unsubscribed
+    ? "Re-subscribe"
+    : "Unsubscribe from all emails"}
+  successBody={unsubscribed
+    ? "You have been re-subscribed to emails"
+    : "You have been unsubscribed from all emails"}
```
[the remark this answers is not in the document]
Comment (scosman) [reply src/routes/(admin)/account/(menu)/settings/change_email_subscription/+page.svelte:23]: I did this one while making some visual tweaks to settings. Those green buttons were bothering me. Other one still outstanding/optional.

File: src/routes/(admin)/account/(menu)/settings/change_email_subscription/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  formTarget="/account/api?/toggleEmailSubscription"
```
[the remark this answers is not in the document]
Comment (evgenyneu) [reply src/routes/(admin)/account/(menu)/settings/change_email_subscription/+page.svelte:24]: Yep, I would prefer an explicit true/false request as well, but went with path of least resistance here :D 
I can I work on this if you are not doing it, and will submit as a separate PR?

**Pull request merged**